module-2/backend/week_two.md

```diff
@@ -4,35 +4,112 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
+7. What is the difference between `#new` and `#create`?<br>
+
+`#new` gives us a blank object with all the attributes being nil. We can use it to check what attributes it has or apparently we can use it as a placeholder for some crazy rails thing called reflection. If we assign the new object to a variable we can fill its attributes and `.save` to make those changes in the database. `#create` does all that at once, makes the object and saves it. 
+
 8. At a basic level, what does cURL allow you to do?
+
+cURL is a command line utitlity that allow you to go to a resource online, grab it and bring it back.
+
 9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
-10. Define foreign key, primary key, and schema.
-11. Describe the relationship between a foreign key on one table and a primary key on another table.
-12. What are the parts of an HTTP response?
-13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
-14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
+
+At its simplest a teacher will have many students and a student belongs to a teacher. In reality perhaps it would be the case of many to many as students have multiple teachers. 
+
+In the simple case the student table would have a foreign key that corresponds to a teacher_id. In a slightly more complex case the teachers and students would be tied through a joins table that would contain id, teacher_id, and student_id. More complex still would be to have the third table have its own attributes, perhaps we could call it curriculum. A teacher would have many students through curriculum and a student would have many teachers through curriculum (curriculum in the sense of program or track). A curriculum would have many students and would have many teachers. 
+
+10. Define foreign key, primary key, and schema.<br>
+
+A foreign key is something_id and will be a part of the table that has the belongs_to. A primary key is a unique identifier(which apparently can be a compound multi-column key if the example calls for it). A schema is a description of how the database is set-up. 
+
+11. Describe the relationship between a foreign key on one table and a primary key on another table.<br>
+
+The foreign key in the one table is the primary key from the other. The table with the foreign key will also have it's primary key.
+
+12. What are the parts of an HTTP response?<br>
+
+I think this is the bit about the three parted thing that rack is based off of. There is the status code and the body(which might be plain ol' html). I'm unable to remember the third part currently.
```

Comment:
An HTTP response has headers and a body. The first header is the status code. Good job remembering the relationship to Rack.